src/main/java/com/google/sps/servlets/GuidesServlet.java

```diff
@@ -23,16 +23,25 @@
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
         response.setContentType("application/json");
-        response.getWriter().print(gson.toJson(helper.queryAll()));
+        String category = request.getParameter("category");
+
+        if(category == null){
+            response.getWriter().print(gson.toJson(helper.queryAll()));
+            return;
+        }
+        
+        category = Jsoup.clean(category, Whitelist.none());
+        response.getWriter().print(gson.toJson(helper.queryByCategory(category)));
     }
 
     @Override
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException{
         String title = Jsoup.clean(request.getParameter("title"), Whitelist.none());
         String description = Jsoup.clean(request.getParameter("description"), Whitelist.none());
         String content = Jsoup.clean(request.getParameter("content"), Whitelist.none());
+        String category = Jsoup.clean(request.getParameter("category"), Whitelist.none());
```

Comment:
Similarly, do you want to store the category as an empty string or null?

Reply:
Well, I am thinking of every parameter to be mandatory, so I should be checking that none of them are null or empty strings and return bad request if any of them are.